vm/x86/jit_x86.cpp

```diff
@@ -1601,6 +1610,31 @@ Compiler::emitThrowPath(int err)
   __ jmp(&report_error_);
 }
 
+void
+Compiler::emitDebugBreakHandler()
+{
+  // Common path for invoking debugger.
+  __ bind(&debug_break_);
+
+  // Get and store the current stack pointer.
+  __ movl(tmp, stk);
+  __ subl(tmp, dat);
+  __ movl(Operand(spAddr()), tmp);
+
+  // Enter the exit frame. This aligns the stack.
+  __ enterExitFrame(ExitFrameType::Helper, 0);
+
+  // Get the context pointer and call the debugging break handler.
+  __ push(0); // IErrorReport*
+  __ push(intptr_t(rt_->GetBaseContext()));
+  __ call(ExternalAddress((void *)InvokeDebugger));

```

Comment:
I believe you need to keep the stack aligned here - it looks like it'll be 8 bytes at the call, when it should be 16. (This matters because GCC assumes an aligned stack and certain SSE instructions will fault if they have an unaligned address.)

Reply:
`enterExitFrame` pushes 3 parameters as well though. I'm confused. Should I still just add
```cpp
__ subl(esp, 8);
```
after calling `enterExitFrame` and do
```cpp
__ addl(esp, 16);
```
after the call?

https://github.com/alliedmodders/sourcepawn/blob/cf5020e8e825fb3647d6e578ff372deb563240c6/vm/x86/macro-assembler-x86.h#L46-L61

I've copied that comment from `emitCallThunk`
https://github.com/alliedmodders/sourcepawn/blob/cf5020e8e825fb3647d6e578ff372deb563240c6/vm/x86/jit_x86.cpp#L1244-L1245